instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/InstrumentationApiContextBridging.java

```diff
@@ -0,0 +1,171 @@
+  private static ContextKeyBridge<Span, io.opentelemetry.api.trace.Span> spanKeyBridge(
+      String name) {
+    try {
+      return new ContextKeyBridge<>(
+          "application.io.opentelemetry.instrumentation.api.internal.SpanKey",
+          "io.opentelemetry.instrumentation.api.internal.SpanKey",
+          name,
+          Bridging::toApplication,
+          Bridging::toAgentOrNull);
+    } catch (Throwable e) {
+      // instrumentation-api may be absent on the classpath, just skip
+      return null;
+    }
+  }
+
+  private static final Class<?> AGENT_HTTP_ROUTE_HOLDER;
+  private static final MethodHandle AGENT_CREATE;
+  private static final MethodHandle AGENT_GET_UPDATED_BY_SOURCE_ORDER;
+  private static final MethodHandle AGENT_GET_ROUTE;
+
+  private static final Class<?> APPLICATION_HTTP_ROUTE_HOLDER;
+  private static final MethodHandle APPLICATION_CREATE;
+  private static final MethodHandle APPLICATION_GET_UPDATED_BY_SOURCE_ORDER;
+  private static final MethodHandle APPLICATION_GET_ROUTE;
+
+  static {
+    MethodHandles.Lookup lookup = MethodHandles.lookup();
+
+    Class<?> agentHttpRouteHolder = null;
+    MethodHandle agentCreate = null;
+    MethodHandle agentGetUpdatedBySourceOrder = null;
+    MethodHandle agentGetRoute = null;
+    Class<?> applicationHttpRouteHolder = null;
+    MethodHandle applicationCreate = null;
+    MethodHandle applicationGetUpdatedBySourceOrder = null;
+    MethodHandle applicationGetRoute = null;
+
+    try {
+      agentHttpRouteHolder =
+          Class.forName("io.opentelemetry.instrumentation.api.instrumenter.http.HttpRouteHolder");
+      agentCreate =
+          lookup.findStatic(
+              agentHttpRouteHolder,
+              "internalCreate",
+              MethodType.methodType(agentHttpRouteHolder, int.class, String.class));
+      agentGetUpdatedBySourceOrder =
+          lookup.findVirtual(
+              agentHttpRouteHolder,
+              "internalGetUpdatedBySourceOrder",
+              MethodType.methodType(int.class));
+      agentGetRoute =
+          lookup.findVirtual(
+              agentHttpRouteHolder, "internalGetRoute", MethodType.methodType(String.class));
+
+      applicationHttpRouteHolder =
+          Class.forName(
+              "application.io.opentelemetry.instrumentation.api.instrumenter.http.HttpRouteHolder");
+      applicationCreate =
+          lookup.findStatic(
+              applicationHttpRouteHolder,
+              "internalCreate",
+              MethodType.methodType(applicationHttpRouteHolder, int.class, String.class));
+      applicationGetUpdatedBySourceOrder =
+          lookup.findVirtual(
+              applicationHttpRouteHolder,
+              "internalGetUpdatedBySourceOrder",
+              MethodType.methodType(int.class));
+      applicationGetRoute =
+          lookup.findVirtual(
+              applicationHttpRouteHolder, "internalGetRoute", MethodType.methodType(String.class));
+    } catch (Throwable ignored) {
```

Comment:
I think something's not quite right. I added `printStackTrace` here and got:

```
java.lang.IllegalAccessException: no such method: io.opentelemetry.javaagent.shaded.instrumentation.api.instrumenter.http.HttpRouteHolder.internalCreate(int,String)HttpRouteHolder/invokeStatic
	at java.base/java.lang.invoke.MemberName.makeAccessException(MemberName.java:972)
	at java.base/java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1117)
	at java.base/java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:3649)
	at java.base/java.lang.invoke.MethodHandles$Lookup.findStatic(MethodHandles.java:2588)
	at io.opentelemetry.javaagent.instrumentation.opentelemetryapi.context.InstrumentationApiContextBridging.<clinit>(InstrumentationApiContextBridging.java:91)
```

Reply:
Now it's all clear: the `lookup()` method is not actually magic and it doesn't magically make things work 😂 

Turns out that the test was incomplete, and only verified one particular scenario (our of many possible).